routing/payment_lifecycle_test.go

```diff
@@ -575,7 +611,7 @@ func TestRouterPaymentStateMachine(t *testing.T) {
 				paymentError,
 			},
 			routes: []*route.Route{
-				shard, shard, shard, shard, shard, shard,
+				shard, shard, shard, shard,
```

Comment:
Why do the last two shards get eliminated?

Reply:
This test actually never uses them, it just needs 4 shards then it succeeds. Pulled this out into a separate commit so it's a bit clearer.